### PR TITLE
RichTextField and RichTextDisplay

### PR DESCRIPTION
**Pull request by guy-ro**
This ended up being super complicated and filled with logic. Apologies in advance for reviewers. (there is a very good reason Github uses markdown). 

To fully understand the code, unfortunately a foundation in Draft.js is useful- contact me if needed! I've added comments for most things a quick dive into Draft.js wouldn't bring up.

**ADDED:**
- Draft.js module to client. Chosen instead of Quill.js for customizability and ease of using with "sanitation".
- RichTextField component (read and write). Supports bold input with a button and key commands. Supports unordered list input with a button and key commands
- RichTextDisplay component (read only, for rendering rich text created by using RichTextField)

**MODIFIED:**
- Seeder now uses the Draft.js format (thus database has requestGroup.description in the correct format)
- RequestGroupDonorView now shows descriptions for requestGroups using RichTextDisplay
- InfoBox modified to have children (so we can easily use RichTextDisplay)
- Moved a bunch of colors around in _config.scss to better reflect how we are using them

image
image
image
image
image

### CODE REVIEW

**Comment by bonnie-chin** [comment]
Wowowow damn okay going the extra mile with Draft.js 😤 thank y'all sm for your work omg 

Some design feedback :D
- Just a lil confused abt the situation here -- after the field goes into its error state the text should be #333333 once they start typing in there 
image

- Super small thing, but I think there's smth up here with the double outline (both a rounded grey box and an unrounded red box)
image

**Review by bonnie-chin — COMMENTED** [review]
Wowowow damn okay going the extra mile with Draft.js 😤 thank y'all sm for your work omg 

Some design feedback :D
- Just a lil confused abt the situation here -- after the field goes into its error state the text should be #333333 once they start typing in there 
image

- Super small thing, but I think there's smth up here with the double outline (both a rounded grey box and an unrounded red box)
image

**Comment by guy-ro** [comment]
Thanks @bonnie-chin ! Removed the double border

**Comment by guy-ro** [comment]
@meganniu regarding

> Some design feedback :D
> 
> * Just a lil confused abt the situation here -- after the field goes into its error state the text should be #333333 once they start typing in there

Should the user of RichTextField set isErroneous to false on change? Or should RichTextField do taht?

**Comment by meganniu** [comment]
@guy-ro 

> Should the user of RichTextField set isErroneous to false on change? Or should RichTextField do taht?

User of RichTextField should be changing isErroneous

File: client/src/components/atoms/RichTextField.tsx [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+            clearState(state)
+            // user is now typing content, so this is active
+            setActive(true)
+            return
+        }
+
+        // check if we are already typing (we are active) and state has no text
+        if (active && !state.getCurrentContent().hasText()) {
+            // if so, we made the content empty, so add back the default text
+            props.onChange(JSON.stringify(convertToRaw(state.getCurrentContent())));
```
Comment (meganniu) [inline client/src/components/atoms/RichTextField.tsx]: I think we need to pass an empty string to props.onChange here
The logic on line 65 and 16 rely on props.initialContent to be falsy/an empty string

File: client/src/components/atoms/RichTextField.tsx [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+            setEditorState(EditorState.createWithContent(ContentState.createFromText(props.defaultText)))
```
Comment (meganniu) [inline client/src/components/atoms/RichTextField.tsx]: I think creating new content means we lose all the history and the user can't undo actions
I'd suggest:
```
llet state = EditorState.createWithContent(ContentState.createFromText(props.defaultText))
if(props.initialContent){
   cosnt tempState = EditorState.createWithContent(convertFromRaw(JSON.parse(props.initialContent)))
   if (tempState.getCurrentContent().hasText()){
    state =  tempState
   } 
}

const [editorState, setEditorState] = React.useState(state) 

function onChange(state: EditorState) {
        // ...

        if (active && !state.getCurrentContent().hasText()) {
            props.onChange(JSON.stringify(convertToRaw(state.getCurrentContent())));
            add props.defaultText to state
            setEditorState(state)
            setActive(false)
            return
        }

        // ...
    }
```
but there are probably better ways to do this

**Review by meganniu — CHANGES_REQUESTED** [review]
Just a few comments, but otherwise looks amazing 🤩🤩
thanks for ur hard work @guy-ro!

**Comment by guy-ro** [comment]
BUG: Empty -> list doesn't work

File: client/src/components/atoms/RichTextField.tsx (hunk 1 above)
Comment (meganniu) [reply client/src/components/atoms/RichTextField.tsx]: nvrm

**Review by meganniu — APPROVED** [review]
LGTM!!!

**Comment by guy-ro** [comment]
@meganniu bugs are fixed

- history is kept on erasing all content
- making a list when content is empty now works

The solution was simple - don't use Draft.js for default text :)

File: client/src/components/atoms/RichTextField.tsx (hunk 2 above)
Comment (guy-ro) [reply client/src/components/atoms/RichTextField.tsx]: fixed

**Comment by meganniu** [comment]
Sweet, everything LGTM! I'll find another reviewer :)))

**Review by Anand-Issac — APPROVED** [review]
Good work, lgtm

**Pull request merged**